contracts/src/Teleporter/ReceiptQueue.sol

```diff
@@ -11,11 +11,13 @@ import "./ITeleporterMessenger.sol";
  * @dev ReceiptQueue is a convenience contract that creates a queue-like interface of
  * TeleporterMessageReceipt structs. It provides FIFO properties.
  */
-contract ReceiptQueue {
-    address public immutable owner;
-    mapping(uint256 => TeleporterMessageReceipt) public queue;
-    uint256 public first = 0;
-    uint256 public last = 0;
+library ReceiptQueue {
+    struct TeleporterMessageReceiptQueue {
+        address owner;
```

Comment:
We shouldn't need to have an "owner" if using this as a library because it is not deployed to it's address with its own storage.

Reply:
is `owner` just used right now for checking whether the receipt queue has been used for a chain yet? Similar to previous `address(receiptQueue) == address(0)`?

Reply:
Can we achieve the same by checking whether size of receipt queue is 0?